File: 0134-gas-station/0134-gas-station.java
class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        int n = gas.length;
        int index = 0;
        int gascount = 0;

        for (int i = 0; i < n; i++) {
            gascount += gas[i] - cost[i];
            if (gascount < 0) {
                index = i + 1;
                gascount = 0;
            }
        }

        int totalgas = 0;
        int totalcost = 0;
        for (int i = 0; i < n; i++) {
            totalgas += gas[i];
            totalcost += cost[i];
        }

        if (totalgas < totalcost)
            return -1;

        return index; 
    }
}

File: 0134-gas-station/0134-gas-station.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
        int n = gas.size();
        int gasCount = 0;
        int start=0;

        for(int i=0;i<n;i++)
        {
            gasCount+=gas[i]-cost[i];
            if(gasCount<0){
                start = i+1;
                gasCount = 0;
            }
        }
        int totalGas=0;
        int totalCost=0;

        for(int i=0;i<n;i++)
        {
            totalGas+=gas[i];
            totalCost+=cost[i];
        }
        if(totalGas<totalCost) return -1;
        return start;
    }
};